std-bits/table/src/main/java/org/enso/table/util/NameDeduplicator.java

```diff
@@ -131,13 +131,19 @@ public List<String> combineWithPrefix(
         output.add(null);
       }
     }
+
     // Second pass - we go over the duplicated names and disambiguate them by adding a prefix and
     // a suffix if necessary.
     for (int i = 0; i < second.size(); i++) {
       String name = second.get(i);
       if (output.get(i) == null) {
-        duplicatedNames.add(name);
-        output.set(i, makeUnique(secondPrefix + name, false));
+        var prefixed = secondPrefix + name;
+        if (isUnique(prefixed)) {
+          output.set(i, prefixed);
+          markUsed(prefixed);
+        } else {
+          output.set(i, makeUnique(secondPrefix + name, false));
+        }
```

Comment:
```suggestion
        var prefixed = secondPrefix + name;
        output.set(i, makeUnique(prefixed, false));
```

Isn't this equivalent? I'm pretty sure it should behave the same, being much simpler.


But then, I think what we actually want to do is:
```suggestion
        var prefixed = secondPrefix + name;
        output.set(i, makeUnique(prefixed, true));
```


That is because, I guess it's ok to not warn about `Right_X` being added. But if `Right_X` is taken and we need to rename to `Right_X_1` - we should warn about duplicate names, right? Setting `reportDuplicate = true` should give us that.